src/Sulu/Component/Content/Repository/ContentRepository.php

```diff
@@ -0,0 +1,636 @@
+            ->addSelect('node', $this->propertyEncoder->localizedContentName('state', $locale), 'state')
+            ->addSelect('node', $this->propertyEncoder->localizedContentName('shadow-on', $locale), 'shadowOn')
+            ->addSelect('node', $this->propertyEncoder->localizedContentName('shadow-base', $locale), 'shadowBase')
+            ->addSelect('node', $this->propertyEncoder->localizedContentName('shadow-base', $locale), 'shadowBase')
+            ->addSelect('node', $this->propertyEncoder->systemName('order'), 'order')
+            ->from($this->qomFactory->selector('node', 'nt:unstructured'))
+            ->orderBy($this->qomFactory->propertyValue('node', 'sulu:order'));
+
+        if (null !== $user) {
+            foreach ($user->getRoleObjects() as $role) {
+                $queryBuilder->addSelect('node', sprintf('sec:%s', 'role-' . $role->getId()), $role->getIdentifier());
+            }
+        }
+
+        return $queryBuilder;
+    }
+
+    /**
+     * Returns array of locales for given webspace key.
+     *
+     * @param string $webspaceKey
+     *
+     * @return string[]
+     */
+    private function getLocalesByWebspaceKey($webspaceKey)
+    {
+        $webspace = $this->webspaceManager->findWebspaceByKey($webspaceKey);
+
+        return array_map(
+            function (Localization $localization) {
+                return $localization->getLocalization();
+            },
+            $webspace->getAllLocalizations()
+        );
+    }
+
+    /**
+     * Returns array of locales for webspaces.
+     *
+     * @return string[]
+     */
+    private function getLocales()
+    {
+        return array_map(
+            function (Localization $localization) {
+                return $localization->getLocalization();
+            },
+            $this->webspaceManager->getAllLocalizations()
+        );
+    }
+
+    /**
+     * Append mapping selects to given query-builder.
+     *
+     * @param QueryBuilder $queryBuilder
+     * @param MappingInterface $mapping Includes array of property names.
+     * @param string[] $locales
+     */
+    private function appendMapping(QueryBuilder $queryBuilder, MappingInterface $mapping, $locales)
+    {
+        $properties = $mapping->getProperties();
+        $properties[] = 'template';
+        $properties[] = 'shadow-on';
+        foreach ($properties as $propertyName) {
+            $this->appendSingleMapping($queryBuilder, $propertyName, $locales);
+        }
+    }
+
+    /**
+     * Append mapping selects for a single property to given query-builder.
+     *
+     * @param QueryBuilder $queryBuilder
+     * @param string $propertyName
+     * @param string[] $locales
+     */
+    private function appendSingleMapping(QueryBuilder $queryBuilder, $propertyName, $locales)
+    {
+        foreach ($locales as $locale) {
+            $alias = sprintf('%s%s', $locale, str_replace('-', '_', ucfirst($propertyName)));
```

Comment:
What does the final alias look like? Why don't you just use the name of the property in PHPCR?